### PR TITLE
POST /thread/set-archive

### LINKED ISSUE #46: POST /thread/set-archive
[Contract](https://pact.serlo.org/pacts/provider/serlo.org/consumer/api.serlo.org/latest#set_%22archived%22_state_of_a_thread_given_there_exists_a_thread_with_a_first_comment_with_an_id_of_100_and_user_with_id_1_is_authenticated)

### PR DESCRIPTION
**Pull request by elbotho**
(closes #46)

### CODE REVIEW

File: src/event/model/event_type.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ pub enum EventType {
     #[serde(rename(
         serialize = "SetThreadStateNotificationEvent",
         deserialize = "discussion/comment/archive",
-        deserialize = "discussion/comment/restore"
+        deserialize = "discussion/restore"
```
Comment (inyono) [inline src/event/model/event_type.rs:10]: @inyono Review the actual semantics of the types. Might be that we don't trigger an event for all possible combinations of threads.

File: src/event/model/event_type.rs [hunk 2]
```diff
@@ -88,7 +88,7 @@ pub enum EventType {
     SetUuidState,
     #[serde(rename(
         serialize = "UnsupportedNotificationEvent",
-        deserialize = "discussion/restore"
+        deserialize = "discussion/comment/restore"
```
Comment (inyono) [inline src/event/model/event_type.rs:91]: This is probably not needed anymore, right?

Comment (elbotho) [reply src/event/model/event_type.rs:91]: to be honest idk: it was added for compatibility with legacy.

**Review by inyono — APPROVED** [review]
Thanks!

**Pull request merged**